### PR TITLE
Add mac version

### PR DESCRIPTION
**Pull request by kdheepak**
New PR based on comments from #20

### CODE REVIEW

**Comment by kdheepak** [comment]
Hmm. This didn't run Travis here either. I'm checking for the build status [here](https://travis-ci.org/kdheepak/OpenDSSDirect.jl/builds/263135397) on my account. If that passes, we can merge.

**Comment by tshort** [comment]
I see failures on mac, and they are failures I haven't seen before.

**Comment by kdheepak** [comment]
It looks like there are segfaults. Hmmm. Is there a easy way for me to run the tests in a more isolated fashion? Like just the tests in one file?

**Comment by tshort** [comment]
To run all tests, do:

```julia
Pkg.test("OpenDSSDirect")
```

To run just one of the test files, do:

```julia
using Base.Test
include(Pkg.dir("OpenDSSDirect", "test", "lowlevel.jl"))
```

Change the "lowlevel.jl" to whichever test you want.

**Comment by kdheepak** [comment]
I cherry picked commits from #16. This passes all tests locally on my machine. Travis results will be [here](https://travis-ci.org/kdheepak/OpenDSSDirect.jl/builds/263183963) when completed.

**Comment by tshort** [comment]
Okay. Instead of commenting that test out, can you comment it out for apple? It's a use case I'd like to see tested. Julia can do this low-level code fast, and I'm interested in experiments here. Something like:

```julia
if !is_apple()
    i = normalsolution_alt()
end
```

**Comment by kdheepak** [comment]
Done! The previous travis build failed on the nightly build though, and I'm not sure why.

**Comment by tshort** [comment]
I'm good with taking out nightly. It seems to cause more problems than it solves.

**Comment by tshort** [comment]
Given that everything but nightly passes on Travi, I'll merge this. Then, I'll remove the nightly tests in a separate commit. Thanks, Dheepak!

**Comment by kdheepak** [comment]
Great! Odd that nightly passes on Linux but not on Mac. I accidentally added a semicolon at the end of the `if` statement, I've fixed that now and pushed those commits again. You can merge after the tests pass on [this page](https://travis-ci.org/kdheepak/OpenDSSDirect.jl/builds/263217822).

**Comment by tshort** [comment]
Everything passed!

**Comment by kdheepak** [comment]
Woohoo!

File: src/core.jl [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 
 if is_windows()
     if Int == Int64
         const dsslib = "$(dirname(@__FILE__))\\..\\deps\\win64\\OpenDSSDirect.DLL"
     else
         const dsslib = "$(dirname(@__FILE__))\\..\\deps\\win32\\OpenDSSDirect.DLL"
     end
-else
-    const dsslib = "$(dirname(@__FILE__))/../deps/libopendssdirect.so"
+elseif is_linux()
```
Comment (tkelman) [inline src/core.jl]: should also check for x86_64 (and ideally that the distro is glibc-based) otherwise this binary isn't going to work

Comment (kdheepak) [inline src/core.jl]: @tkelman To my understanding there's no x86 compiler for fpc for Linux, so this will likely only work in x64 environments on Linux.

Comment (tkelman) [inline src/core.jl]: Right but Julia supports non-glibc Linux distributions, as well as 32 bit x86, 32 and 64 bit ARM, and 64 bit Power. `is_linux()` will be true for all of them.

Comment (kdheepak) [inline src/core.jl]: I see. You should open a new issue regarding this.

Comment (tshort) [inline src/core.jl]: @tkelman, that's why we need a BinDeps2! 😈  I cringe at including binary libraries as it is.

I'm trying out a fix for this on Travis now. That still doesn't help for nonstandard platforms where someone may compile the library themselves. But, hopefully, we'll improve our library build process.

Comment (tkelman) [inline src/core.jl]: Checking binaries in to git isn't generally a good idea. If someone is using an exotic platform usually they're on their own, but if they're able to get the library compiled, they shouldn't have to modify git-tracked files to override your default binary.

Comment (tshort) [inline src/core.jl]: I agree. It was the path of least resistance. Hopefully temporary.

Comment (tkelman) [inline src/core.jl]: Attaching binaries to github releases is quite simple.

Comment (tshort) [inline src/core.jl]: Good to know. I assumed we'd have to host them somewhere else.

**Pull request merged**